rust/src/encodings/plain.rs

```diff
@@ -178,7 +186,7 @@ impl<'a> PlainDecoder<'a> {
             )));
         }
         let start_offset = make_byte_offset(self.data_type, start)?;
-        let end_offset = make_byte_offset(self.data_type, end)?;
+        let end_offset = make_ceil_byte_offset(self.data_type, end)?;
```

Comment:
does integer division always round down? do we need a `make_floor_byte_offset` for start_offset?

Reply:
yes it should be, given that offset are unsigned int?